app/synchronizationmanager.cpp

```diff
@@ -9,81 +9,165 @@
 
 #include "synchronizationmanager.h"
 
-SynchronizationManager::SynchronizationManager( MerginApi *backend, QObject *parent )
+SynchronizationManager::SynchronizationManager(
+  MerginApi *merginApi,
+  QObject *parent
+)
   : QObject( parent )
-  , mAutosyncController( nullptr )
-  , mBackend( backend )
+  , mMerginApi( merginApi )
 {
-  QObject::connect( mBackend, &MerginApi::syncProjectFinished, this, &SynchronizationManager::syncProjectFinished );
-  QObject::connect( mBackend, &MerginApi::syncProjectStatusChanged, this, &SynchronizationManager::syncProjectStatusChanged );
+  if ( mMerginApi )
+  {
+    QObject::connect( mMerginApi, &MerginApi::pushCanceled, this, &SynchronizationManager::onProjectSyncCanceled );
+    QObject::connect( mMerginApi, &MerginApi::syncProjectFinished, this, &SynchronizationManager::onProjectSyncFinished );
+    QObject::connect( mMerginApi, &MerginApi::syncProjectStatusChanged, this, &SynchronizationManager::onProjectSyncProgressChanged );
+  }
 }
 
-SynchronizationManager::~SynchronizationManager()
+SynchronizationManager::~SynchronizationManager() = default;
+
+void SynchronizationManager::syncProject( const Project &project, bool withAuth )
```

Comment:
so the SynchronizationTransaction is not returned? or it will be added later? (as it is in the schema )

Reply:
No, it is not returned, synchronization status is read directly from `SynchronizationManager`. SynchronizationManager only informs ProjectsModel that some data has changed. SynchronizationManager holds all the sync data.

Reply:
So the schema is not updated?